pkg/cache/resource.go

```diff
@@ -0,0 +1,78 @@
+package cache
+
+import (
+	corev1 "k8s.io/api/core/v1"
+
+	"sigs.k8s.io/kueue/pkg/resources"
+)
+
+type resourceGroupNode interface {
+	getResourceGroups() *[]ResourceGroup
+}
+
+// resourceGroupForResource returns the ResourceGroup which contains capacity
+// for the resource, or nil if the CQ doesn't provide this resource.
+func resourceGroupForResource(node resourceGroupNode, resource corev1.ResourceName) *ResourceGroup {
+	resourceGroups := node.getResourceGroups()
+	for i := range *resourceGroups {
+		if (*resourceGroups)[i].CoveredResources.Has(resource) {
+			return &(*resourceGroups)[i]
+		}
+	}
+	return nil
+}
+
+// flavorResources returns all of the FlavorResource(s) defined in the given node.
+// note: this is copy heavy, but we don't care - the structure of ResourceGroup will change
+// soon, which will have these values precomputed (or cheap to access).
+func flavorResources(node resourceGroupNode) []resources.FlavorResource {
+	flavorResources := make([]resources.FlavorResource, 0)
```

Comment:
```suggestion
	var flavorResources []resources.FlavorResource
```

`append` can allocate as necessary.